### ISSUE TITLE
Nightly build failures: error: ‘int Kokkos::Impl::{anonymous}::get_gpu(const Kokkos::InitArguments&)’ defined but not used

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#3071]
Following merge of PR #3061, nightly tests began failing with `-Werror`

List of failing builds here:
https://jenkins-son.sandia.gov/view/Kokkos/

Sample failing build used for error message and reproducers:
https://jenkins-son.sandia.gov/job/Kokkos_develop_kokkosdev2_gcc830_openmp_serial/134/console

**Error message**:
```bash
21:36:08 /home/jenkins/slave/workspace/Kokkos_develop_kokkosdev2_gcc830_openmp_serial/kokkos/core/src/impl/Kokkos_Core.cpp:171:5: error: ‘int Kokkos::Impl::{anonymous}::get_gpu(const Kokkos::InitArguments&)’ defined but not used [-Werror=unused-function]
21:36:08  int get_gpu(const InitArguments& args) {
21:36:08      ^~~~~~~
21:36:08 cc1plus: all warnings being treated as errors
```

**Sample reproducer** (gcc/8.3 OpenMP_Serial build):
```bash
  #   Load modules:
        module load sems-env kokkos-env sems-cmake/3.12.2 kokkos-hwloc/1.10.1/base sems-gcc/8.3.0

        $KOKKOS_PATH/generate_makefile.bash --with-devices=OpenMP --arch=SNB,Volta70 --compiler=/projects/sems/install/rhel7-x86_64/sems/compiler/gcc/8.3.0/base/bin/g++ --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized " --cxxstandard="11" --ldflags=""  --kokkos-path=$KOKKOS_PATH --with-options=disable_deprecated_code --with-cuda-options= --no-examples 
```

**Comment by ndellingwood** [comment]
@dalg24 is `-Werror` used in any of the CI builds?
These are the warning flags used in most of the nightly builds, would help to determine the discrepancy to prevent this type of issue slipping through:
`-Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized`

**Comment by ndellingwood** [comment]
Adding @crtrott @masterleinad @bartlettroscoe based on involvement from #3061

**Comment by masterleinad** [comment]
We are using `-Werror` on all `jenkins` builds and the default warnings
`-Wall -Wunused-parameter -Wshadow -pedantic -Wsign-compare -Wtype-limits -Wuninitialized` for `nvcc` and `clang` and `-Wempty-body -Wclobbered -Wignored-qualifiers -Wall -Wunused-parameter -Wshadow -pedantic -Wsign-compare -Wtype-limits -Wuninitialized` for `gcc`.

We don't enable compiler warnings (explicutly) for `Travis` (but use `-Werror`) and `appveyor`, though, AFAICT.

**Comment by bartlettroscoe** [comment]
Is this warning occurring in non-CUDA builds?

**Comment by ndellingwood** [comment]
These errors occurred in non-CUDA builds. Resolved by PR #3072
